docker-compose-rule-junit4/src/main/java/com/palantir/docker/compose/DockerComposeRule.java

```diff
@@ -159,8 +158,11 @@ public void before() throws IOException, InterruptedException {
     @Override
     public void after() {
         try {
-            shutdownStrategy().shutdown(this.dockerCompose(), this.docker());
-            logCollector().stopCollecting();
+            shutdownStrategy().stop(this.dockerCompose());
+
+            logCollector().collectLogs(this.dockerCompose());
```

Comment:
Also, on another very important note, if cluster fails to start up in `before()`, `after()` will not be run (see `ExternalResource` and how bad it is). I think this is indicative that we should add an integTest to check that logs are collected if the cluster fails to start up, and for this PR when the container is restarted, just to make sure we maintain these behaviours in the future.

To fix the `ExternalResource` problem, the option is to stop using `ExternalResource` and write our own apply method that puts the `before` call in the `try` block. This is probably the behaviour we always wanted anyway.